engines/metaengine.h

```diff
@@ -54,33 +54,51 @@ class GuiObject;
 class OptionsContainerWidget;
 }
 
+/**
+ * @defgroup engines_metaengine Meta engine
+ * @ingroup engines
+ *
+ * @brief API for managing various functions of the meta engine.
+ *
+ * @{
+ */
+
 /**
  * Per-game extra GUI options structure.
  * Currently, this can only be used for options with checkboxes.
  */
 struct ExtraGuiOption {
-	const char *label;          // option label, e.g. "Fullscreen mode"
-	const char *tooltip;        // option tooltip (when the mouse hovers above it)
-	const char *configOption;   // confMan key, e.g. "fullscreen"
-	bool defaultState;          // the detault state of the checkbox (checked or not)
+	const char *label;         /*!< Option label, e.g. "Fullscreen mode". */
+	const char *tooltip;       /*!< Option tooltip shown when the mouse cursor hovers over it. */
+	const char *configOption;  /*!< confMan key, e.g. "fullscreen". */
+	bool defaultState;         /*!< Default state of the checkbox (checked or not). */
 };
 
+/**
+ * Array of ExtraGuiOption structures.
+ */
 typedef Common::Array<ExtraGuiOption> ExtraGuiOptions;
 
+/**
+ * @todo Doc required
+ */
 enum { kSavegameFilePattern = -99 };
 
 #define EXTENDED_SAVE_VERSION 4
 
+/**
+ * Structure describing a savegame file.
+ */
 struct ExtendedSavegameHeader {
-	char id[6];
-	uint8 version;
-	Common::String saveName;
-	Common::String description;
-	uint32 date;
-	uint16 time;
-	uint32 playtime;
-	Graphics::Surface *thumbnail;
-	bool isAutosave;
+	char id[6];                   /*!< ID of the savegame file. */
+	uint8 version;                /*!< Version of the game. */
```

Comment:
This is actually the version for the save game header and not the game version.

The idea here is that we may change the content of this header from time to time. For example the initial header we were using in old ScummVM versions did not have the `description` and `isAutoSave` information. When we read back those savegame files we need to know what header information is in the file so that we can read it properly. So the first thing that is read is the version, and then we know what else is there and read the rest of the header.